include/singa/core/memory.h

```diff
@@ -50,6 +51,57 @@ class DeviceMemPool {
 //  size_t init_size_ = 0, max_size_ = 0;
 };
 
+class CppMemPool {
+	public:
+		// initial pool size (MB), and the size of each memory uint in the memory pool (KB)
+		CppMemPool(size_t init_size_mb = 256, size_t uint_size_kb = 1);
```

Comment:
Are all memory units of the same size?
Can we do it in this way:
1. create a large memory pool P0 at the beginning
2. when malloc is called, return a block of the asked size and move the offset of the start pointer of the pool Pi
3. if not enough space for malloc a new block, then create a new block Pi+1.